### DISCUSSION TITLE
Add clear message of error, for method Country::getByIso()

### DISCUSSION THREAD

**Discussion by Zudjo** [PrestaShop/PrestaShop#33981] [Ideas]
### The problem

I lost a good hour trying to understand where the "Fatal Error" without any explanation was coming from...

### The solution

```
    public static function getByIso($isoCode, $active = false)
    {
        if (!Validate::isLanguageIsoCode($isoCode)) {
            die(Tools::displayError('Given iso code (' . $isoCode . ') is not valid.')); // Just add a message like this
        }
        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow(
            '
			SELECT `id_country`
			FROM `' . _DB_PREFIX_ . 'country`
			WHERE `iso_code` = \'' . pSQL(strtoupper($isoCode)) . '\''
            . ($active ? ' AND active = 1' : '')
        );

        if (isset($result['id_country'])) {
            return (int) $result['id_country'];
        }

        return false;
    }
```

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @Zudjo 

Could you submit a pull request for the 8.1.x branch with the proposed solution? :-)

**Comment by DMartinezCC** [comment]
Hello.

Throwing a "die" into a method that serves to obtain an ISO is absolute madness.

Throwing an exception should be enough. Returning false should also be sufficient. Forcing validation prior to launching the method, so that the method then validates again does not make sense.

Calling this method is very common when creating addresses. Which are usually related to orders. Killing execution in mass order creation is a serious problem

Please consider changing that die
